packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,210 @@
+  | BAR plotData                                                { yy.setBarData({text: '', type: 'text'}, $plotData); }
+  | BAR text plotData                                           { yy.setBarData($text, $plotData); }
+  ;
+
+plotData
+  : SQUARE_BRACES_START commaSeperateNumber SQUARE_BRACES_END   { $$ = $commaSeperateNumber }
+  ;
+
+commaSeperateNumber
+  : NUMBER_WITH_DECIMAL commaSeperateNumberValues                 {
+                                                                    $commaSeperateNumberValues = $commaSeperateNumberValues || [];
+                                                                    $commaSeperateNumberValues.unshift(Number($NUMBER_WITH_DECIMAL));
+                                                                    $$ = $commaSeperateNumberValues
+                                                                  }
+  ;
+
+commaSeperateNumberValues
+  : COMMA NUMBER_WITH_DECIMAL commaSeperateNumberValues           {
+                                                                    $commaSeperateNumberValues = $commaSeperateNumberValues || [];
+                                                                    $commaSeperateNumberValues.unshift(Number($NUMBER_WITH_DECIMAL));
+                                                                    $$ = $commaSeperateNumberValues
+                                                                  }
+  |;
+
+parseXAxis
+  : text                                                          {yy.setXAxisTitle($text);}
+  | text bandData                                                 {yy.setXAxisTitle($text); yy.setXAxisBand($bandData);}
+  | text NUMBER_WITH_DECIMAL ARROW_DELIMITER NUMBER_WITH_DECIMAL  {yy.setXAxisTitle($text); yy.setXAxisRangeData(Number($NUMBER_WITH_DECIMAL1), Number($NUMBER_WITH_DECIMAL2));}
+  ;
+
+bandData
+  : SQUARE_BRACES_START commaSeperateText SQUARE_BRACES_END       {$$ = $commaSeperateText}
+  ;
+
+commaSeperateText
+  : text commaSeperateTextValues                                  {
+                                                                    $commaSeperateTextValues = $commaSeperateTextValues || [];
+                                                                    $commaSeperateTextValues.unshift($text);
+                                                                    $$ = $commaSeperateTextValues
+                                                                  }
+  ;
+
+commaSeperateTextValues
+  : COMMA text commaSeperateTextValues                            {
+                                                                    $commaSeperateTextValues = $commaSeperateTextValues || [];
+                                                                    $commaSeperateTextValues.unshift($text);
+                                                                    $$ = $commaSeperateTextValues
+                                                                  }
+  |;
```

Comment:
You could've united these two:
```
commaSeparatedTexts
  : text COMMA commaSeparatedTexts { $$ = [text, ...$commaSeparatedTexts] }
  | text { $$ = [text] }
  ;
```

Again, I have not executed this code, needs verifying.

Your version will be ok too, this one is a little bit shorter, though.

Reply:
Done